### PR TITLE
add unit_tests.yml

### PR DESCRIPTION
**Pull request by masenf**

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+env:
+  OPENAI_API_KEY: dummy
+  TELEMETRY_ENABLED: false
+  REFLEX_DEP: "git+https://github.com/reflex-dev/reflex@main"
+
+on:
+  push:
+    branches: ['main']
+    paths-ignore:
```
Comment (tgberkeley) [inline .github/workflows/unit_tests.yml:15]: Why are we ignoring markdown files for this GitHub workflow? As these can also contain valid or invalid urls.

File: docs/datatable_tutorial/live_stream.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ darkTheme = {
...
 
-Here we use a [Background Task](https://reflex.dev/docs/events/background_events/) to stream the data into the table without blocking UI interactivity. We call an advice API using `httpx` and then append that data to the `self.table_data` state var. We also create a button that allows us to start and pause the streaming of the data by changing the value of the boolean state var `running` using the event handler `toggle_pause`. If the `running` state var is set to `True` we stream the API data, when it is set to `False` we break out of the `while` loop and end the background event.
+Here we use a [Background Task](https://reflex.dev/docs/events/background-events/) to stream the data into the table without blocking UI interactivity. We call an advice API using `httpx` and then append that data to the `self.table_data` state var. We also create a button that allows us to start and pause the streaming of the data by changing the value of the boolean state var `running` using the event handler `toggle_pause`. If the `running` state var is set to `True` we stream the API data, when it is set to `False` we break out of the `while` loop and end the background event.
```
Comment (picklelo) [inline docs/datatable_tutorial/live_stream.md:37]: We should convert this to a path instead of hardcoding

File: tests/test_urls.py [hunk 3]
```diff
@@ -18,16 +18,23 @@ def check_urls(repo_dir):
     """
     url_pattern = re.compile(r'http[s]?://reflex\.dev[^\s")]*')
     errors = []
-
+    print(repo_dir)
```
Comment (picklelo) [inline tests/test_urls.py:21]: remove?

File: docs/datatable_tutorial/live_stream.md (hunk 2 above)
Comment (tgberkeley) [reply docs/datatable_tutorial/live_stream.md:37]: changed

File: tests/test_urls.py (hunk 3 above)
Comment (tgberkeley) [reply tests/test_urls.py]: removed

**Review by picklelo — APPROVED** [review]

**Pull request merged**